### PR TITLE
Enhance customization of house and counselor for SHS and Full Circle.

### PR DESCRIPTION
**Pull request by jhilde**
This PR implements the change for #1208 making #1201 better for the end user.

When I added the house and counselor fields for Somerville High School, we were a bit wary of adding too many district-specific and school-specific attributes. That still stands, but for the moment we're being very cautious about it.

I added house and counselor conditionally for all schools that are high schools. At the time, the only one was SHS. As we add Full Circle (Somerville's alternative high school), we need to better support the fields it needs. Namely, Full Circle does not have houses but does have counselors.

Rather than replicating the logic in each place where that data is exposed, I decided to create a simple helper in `app/assets/javascripts/helpers/customization_helper.js` There's currently one function `shouldDisplay` that takes an `element` and a `school`.  

It's pretty conservative in that it returns false for elements it doesn't know about, for a null school, or for missing attributes on school.

This is a very small step toward encapsulating this customization logic. As we discover more about future districts, we will handle this differently. But for now this is a first step.

### CODE REVIEW

File: app/assets/javascripts/helpers/customization_helpers.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+function shouldDisplay(element, schoolContext) {
```
Comment (alexsoble) [inline app/assets/javascripts/helpers/customization_helpers.js]: Oooooh I like this approach! If I'm reading it right this is similar to the the "centralize decision-making methods into one place" pattern that @kevinrobinson has talked about for authorization.

File: app/assets/javascripts/helpers/customization_helpers.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      return "SHS" === schoolContext.local_id;
+    }
+    return false;
+  case 'counselor': 
+    if (schoolContext && schoolContext.school_type) {
+      return "HS" === schoolContext.school_type;
+    }
+    return false;
+  }
+  return false;
```
Comment (alexsoble) [inline app/assets/javascripts/helpers/customization_helpers.js]: What is this second layer of `return false` doing? Is it kind of like a default?

Comment (jhilde) [reply app/assets/javascripts/helpers/customization_helpers.js]: Yup, forgot about that default thing. :-)

File: app/assets/javascripts/helpers/customization_helpers.js (hunk 1 above)
Comment (jhilde) [reply app/assets/javascripts/helpers/customization_helpers.js]: Yeah, depending on customization needs, we likely need a more robust solution that lives in a data store, not code. But for the moment this gets us a step in a good direction.

Comment (kevinrobinson) [reply app/assets/javascripts/helpers/customization_helpers.js]: 👍 👞 👠 <--- those are steps :)

File: app/assets/javascripts/helpers/customization_helpers.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+function shouldDisplay(element, schoolContext) {
+  switch(element) {
+  case 'house':
+    if (schoolContext && schoolContext.local_id) {
+      return "SHS" === schoolContext.local_id;
+    }
+    return false;
+  case 'counselor': 
+    if (schoolContext && schoolContext.school_type) {
```
Comment (kevinrobinson) [inline app/assets/javascripts/helpers/customization_helpers.js]: Is `schoolContext` the same as `school`?

**Comment by alexsoble** [comment]
Hey there @jhilde! #1204 is ready to merge/ship. On our call last week it would be best to merge this PR in first. Looks like @kevinrobinson has a comment open. Let's prioritize merging this in over refactoring work so that we can unblock turning on Capuano and Full Circle.

**Comment by jhilde** [comment]
Changed the name in response to @kevinrobinson's comment. Made sense in my mind at the time, not sure why. Keep it simple!

Will deploy @alexsoble once build passes.

**Comment by jhilde** [comment]
🚢 
✔️ ✔️  on ie11

**Comment by alexsoble** [comment]
👍 👍  Thanks @jhilde!

**Pull request merged**